pyglossary/plugins/ebook_kobo.py

```diff
@@ -179,51 +170,46 @@ def writeGroup(lastPrefix):
 				dataEntryCount += 1
 				continue
 			l_word = entry.l_word
-			if len(l_word) == 1:
-				data.append(entry.getRaw(glos))
-				continue
+			allWords += l_word
 			wordsByPrefix = OrderedDict()
 			for word in l_word:
 				prefix = self.get_prefix(word)
 				if prefix in wordsByPrefix:
 					wordsByPrefix[prefix].append(word)
 				else:
 					wordsByPrefix[prefix] = [word]
-			if len(wordsByPrefix) == 1:
-				data.append(entry.getRaw(glos))
-				continue
-			defi = entry.defi
+			defi = self.fix_defi(entry.defi)
+			mainHeadword = l_word[0]
 			for prefix, p_words in wordsByPrefix.items():
-				data.append(Entry(p_words, defi).getRaw(glos))
+				headword, *variants = p_words
+				if headword != mainHeadword:
+					headword = f"{mainHeadword}, {headword}"
+				data.append(
+					compress(dumps((prefix, headword, variants, defi)), level=9)
```

Comment:
Most compression algorithms need larger amounts of data to get really effective. The compression here only decreased the memory consumption by 5% in my tests while increasing the run time by more than 10%. I would just keep it simple and remove the compression. The memory usage is totally reasonable IMO (~60MB for medium dictionaries in my case).

Reply:
Yeah I know, but I still rather keep it. I can try to improve it later. Glossaries can be much bigger and cause a crash which is not good.